tests/www/approvals_views/test_detail.py

```diff
@@ -24,3 +43,404 @@ def test_redirect(self, client):
+            client.force_login(user)
+            response = client.get(url)
+            approval_box = parse_response_to_soup(response, selector=".c-box--pass")
+            assert str(approval_box) == snapshot(name="suspended approval")
+        suspension.delete()
+
+        # Expired
+        approval.end_at = approval.start_at + datetime.timedelta(days=1)
+        approval.save(update_fields=("end_at",))
+        for user in (employer, prescriber):
+            client.force_login(user)
+            response = client.get(url)
+            approval_box = parse_response_to_soup(response, selector=".c-box--pass")
+            assert str(approval_box) == snapshot(name="expired approval")
+
+        # Future
+        approval.start_at = datetime.date(2025, 1, 1)
+        approval.end_at = datetime.date(2025, 1, 2)
+        approval.save(update_fields=("start_at", "end_at"))
+        for user in (employer, prescriber):
+            client.force_login(user)
+            response = client.get(url)
+            approval_box = parse_response_to_soup(response, selector=".c-box--pass")
+            assert str(approval_box) == snapshot(name="future approval")
+
+    @freeze_time("2024-09-25", tick=True)  # tick is important for job applications' created_by
+    def test_approval_detail_with_suspensions(self, client, snapshot):
+        job_application = JobApplicationFactory(
+            to_company__name="Mon SIAE",
+            state=JobApplicationState.ACCEPTED,
+            with_approval=True,
+        )
+        approval = job_application.approval
+        url = reverse("approvals:details", kwargs={"pk": approval.pk})
+        employer = job_application.to_company.members.first()
+        prescriber = job_application.sender
+        assert prescriber.is_prescriber
+
+        # Valid
+        current_suspension = SuspensionFactory(
+            id=1,
+            approval=approval,
+            start_at=timezone.localdate() - relativedelta(days=7),
+            end_at=timezone.localdate() + relativedelta(days=3),
+            siae__name="Une SIAE",
+        )
+        # Older
+        SuspensionFactory(
+            id=2,
+            approval=approval,
+            start_at=timezone.localdate() - relativedelta(days=30),
+            end_at=timezone.localdate() - relativedelta(days=20),
+            siae__name="Une autre SIAE",
+        )
+        suspension_update_url = reverse("approvals:suspension_update", kwargs={"suspension_id": current_suspension.pk})
+        suspension_delete_url = reverse(
+            "approvals:suspension_action_choice", kwargs={"suspension_id": current_suspension.pk}
+        )
+
+        def get_suspensions_section(response):
+            return parse_response_to_soup(
+                response,
+                selector="#suspensions-list",
+                replace_in_attr=[
+                    ("href", f"/approvals/details/{approval.pk}", "/approvals/details/[PK of Approval]"),
```

Comment:
Je crois que tu peux juste passer `replace_in_attr=[approval]`

Reply:
Dans l'idée oui. Mais comme les PK des approvals sont des entiers faire des `.replace("1", "[PK of Approval]`)` dans toutes les href a de bonne chance de ne pas faire ce qu'on veut :sweat_smile: .

Reply:
Certes ^^'